pasta_eln/GUI/configSetupLinux.py

```diff
@@ -57,91 +60,91 @@ def callbackProgress(self, number:int) -> None:
-      if button == QMessageBox.Yes:
-        dirName = QFileDialog.getExistingDirectory(self,'Create and select directory for scientific data',str(Path.home()))
-        installLinuxRoot(existsCouchDB, Path(dirName))
-        logging.info('Install linux root finished')
-      else:
-        self.mainText = self.mainText.replace('- CouchDB','- CouchDB: user chose to NOT install' )
-        self.text1.setMarkdown(self.mainText)
+    if command[0] is Command.ANALYSE:
+      flagContinue = True
+      logging.info('Linux setup analyse start')
 
-    #Configuration
-    if flagContinue:
-      res = configuration('test')
+      #Couchdb
+      existsCouchDB = None
+      res = couchdb('test')
       if res =='':
-        self.mainText = self.mainText.replace('- Configuration of preferences','- Configuration of preferences is acceptable' )
+        self.mainText = self.mainText.replace('- CouchDB','- CouchDB is installed' )
         self.text1.setMarkdown(self.mainText)
+        existsCouchDB = True
       else:
-        button = QMessageBox.question(self, "PASTA-ELN configuration", "Do you want to create/repair the configuration.")
+        existsCouchDB = False
+
+      #Install couchdb
+      if not existsCouchDB:
+        button = QMessageBox.question(self, "Root installations", rootInstallLinux)
         if button == QMessageBox.Yes:
-          configuration('repair')
+          dirName = QFileDialog.getExistingDirectory(self,'Create and select directory for scientific data',str(Path.home()))
+          installLinuxRoot(existsCouchDB, Path(dirName))
+          logging.info('Install linux root finished')
         else:
-          self.mainText = self.mainText.replace('- Configuration of preferences','- Configuration: user chose to NOT install' )
+          self.mainText = self.mainText.replace('- CouchDB','- CouchDB: user chose to NOT install' )
           self.text1.setMarkdown(self.mainText)
-          flagContinue = False
 
-    #Ontology
-    if flagContinue:
-      res = ontology('test')
-      if '**ERROR' not in res:
-        self.mainText = self.mainText.replace('- Ontology of the datastructure','- Ontology of the datastructure is acceptable\n'+res )
+      #Configuration
+      if flagContinue:
+        res = configuration('test')
+        if res =='':
+          self.mainText = self.mainText.replace('- Configuration of preferences','- Configuration of preferences is acceptable' )
+          self.text1.setMarkdown(self.mainText)
+        else:
+          button = QMessageBox.question(self, "PASTA-ELN configuration", "Do you want to create/repair the configuration.")
+          if button == QMessageBox.Yes:
+            configuration('repair')
+          else:
+            self.mainText = self.mainText.replace('- Configuration of preferences','- Configuration: user chose to NOT install' )
+            self.text1.setMarkdown(self.mainText)
+            flagContinue = False
+
+      #Ontology
+      if flagContinue:
```

Comment:
Can't we merge this with the loop above? .. also all the subsequent if loops can be merged?

Reply:
One could merge, but it helps the readability if things are happening subsequently, also for the flagContinue idea: if user wants to break, she/he can do so. Also, this is a bug-fix and not a rewrite PR: improve the code to remove the bug.